### PR TITLE
Feature/robot faker

### PR DESCRIPTION
**Pull request by boakley**
The core of this PR are two new keywords in Salesforce.py, `Get Fake Data` and `Set Faker Locale`. In addition, the `__init__` method of the Salesforce class has been updated to create a new global variable, `${faker}`. The keywords themselves have a lot of documentation, but more examples can be found in the quip doc titled "Faker: A new way to generate test data". 

This includes an additional robot test suite faker.robot, updates to the create_contact.robot test and several other tests that were previously using random data, and an updated robotframework.rst and Keywords.html files. 

# Critical Changes

# Changes
* Two new keywords: `Get Fake Data` and `Set Faker Locale`
* This adds a new global robot variable named ${Faker}. 

# Issues Closed

### CODE REVIEW

File: cumulusci/robotframework/Salesforce.py [hunk 1]
```diff
@@ -124,6 +132,80 @@ def _jsclick(self, locator):
...
+        Example
+
+        | Set Faker Locale    fr_FR
+        | ${french_address}=  Faker  address
+
+        """
+        try:
+            self._faker = faker.Faker(locale)
+        except AttributeError:
+            raise Exception(f"Unknown locale for fake data: '{locale}'")
```
[the remark this answers is not in the document]
Comment (boakley) [reply cumulusci/robotframework/Salesforce.py:154]: No, that's not required. There's a test case that covers this -- it switches from fr_FR to it_IT and then gets a VAT id which has a country-specific prefix.

File: cumulusci/robotframework/Salesforce.py [hunk 2]
```diff
@@ -124,6 +132,80 @@ def _jsclick(self, locator):
...
+
+        The return value is typically a string, though in some cases
+        some other type of object will be returned. For example, the
+        ``date_between`` fake returns a
+        [https://docs.python.org/3/library/datetime.html#date-objects|datetime.date
+        object]. Each time a piece of fake data is requested it will
+        be regenerated, so that multiple calls will usually return
+        different data.
+
+        This keyword can also be called using robot's extend variable
```
[the remark this answers is not in the document]
Comment (boakley) [reply cumulusci/robotframework/Salesforce.py:180]: Thanks. Fixed.

File: cumulusci/robotframework/Salesforce.py [hunk 3]
```diff
@@ -124,6 +132,80 @@ def _jsclick(self, locator):
...
+        syntax using the variable ``${faker}``. In such a case, the
+        data being asked for is a method call and arguments must be
+        enclosed in parenthesis and be quoted. Arguments should not be
```
[the remark this answers is not in the document]
Comment (boakley) [reply cumulusci/robotframework/Salesforce.py:183]: Thanks. Fixed.

File: cumulusci/robotframework/Salesforce.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci/robotframework/Salesforce.py:154]: Huh. I see the passing test, but I don't understand how it works!

File: cumulusci/robotframework/tests/salesforce/faker.robot [hunk 4]
```diff
@@ -0,0 +1,67 @@
+*** Settings ***
+Resource  cumulusci/robotframework/Salesforce.robot
```
Comment (davisagli) [inline cumulusci/robotframework/tests/salesforce/faker.robot:2]: This suite should probably use `Force Tags  no-browser` so it runs in the CI job for tests that don't require a browser.

Comment (boakley) [reply cumulusci/robotframework/tests/salesforce/faker.robot:2]: Done.

**Pull request merged**